docs/website/blog/2022-05-02-odo-v3-alpha1.md

```diff
@@ -0,0 +1,327 @@
+      kind: build
+    workingDir: /project
+- id: run
+  exec:
+    commandLine: npm start
+    component: runtime
+    group:
+      isDefault: true
+      kind: run
+    workingDir: /project
+- id: debug
+  exec:
+    commandLine: npm run debug
+    component: runtime
+    group:
+      isDefault: true
+      kind: debug
+    workingDir: /project
+- id: test
+  exec:
+    commandLine: npm test
+    component: runtime
+    group:
+      isDefault: true
+      kind: test
+    workingDir: /project
+- id: deploy
+  composite:
+    commands:
+    - build-image
+    - k8s-deployment
+    - k8s-service
+    group:
+      isDefault: true
+      kind: deploy
+- id: build-image
+  apply:
+    component: outerloop-build
+- id: k8s-deployment
+  apply:
+    component: outerloop-deployment
+- id: k8s-service
+  apply:
+    component: outerloop-service
+starterProjects:
+- name: nodejs-starter
+  git:
+    remotes:
+      origin: https://github.com/odo-devfiles/nodejs-ex.git
+```
+
+The `runtime` component defines the container that will be deployed to support the program in development. Specifically, it will use the image `registry.access.redhat.com/ubi8/nodejs-14:latest`, and sources should be placed in the `/project` directory of the container. Two endpoints are also defined, one to access the micro-service, the other to help the debugger attach to the process, during debugging sessions.
+
+The commands `install`, `run`, `debug` and `test` indicate which commands to execute to respectively build, execute, debug and test the application. For example, the `npm install` command will be executed in the container to build the application, then `npm start` will be executed to start the application.
+
+To deploy the micro-service, the component `outerloop-build` indicates how to build the production image (by using `./Dockerfile`, and creating an image whose name is defined by the variable `CONTAINER_IMAGE` defined at the beginning of the devfile). then, two other components `outerloop-deployment` and `outerloop-service` define the Kubernetes resources to deploy to the cluster. Note that the first one defines a Deployment that will help deploy a container using the image built with the previous `outerloop-build` component.
+
+The `starterProjects` section at the end of the Devfile indicates a list of starter projects, that can be downloaded to have an example of program deployable with this Devfile.
+
+## Devfile registry
+
+We can see through the previous example that a Devfile is generic enough, with only a few specific values, like the endpoints and the image names. A Devfile written for a specific language and framework can be used by most of the programs written using this language and framework, with minimum personalization.
+
+A Devfile registry is available at https://registry.devfile.io, containing Devfile's for a large variety of languages and frameworks, and you can deploy your own registry to make accessible your own Devfile's.
```

Comment:
````suggestion
A Devfile registry is available at https://registry.devfile.io, containing Devfiles for a large variety of languages and frameworks, and you can deploy your own registry to make accessible your own Devfiles.
```

Reply:
`Devfile's` is the notation @cdrage has used in his review. I think he didn't want to add an `s` to an invariant name

Reply:
What about this one?

Reply:
I'll use Devfiles instead: https://createteachinspire.com/proper-nouns-plural-possessive/